Sources/Sentry/SentryProfiler.mm

```diff
@@ -93,12 +93,19 @@ @implementation SentryProfiler {
     uint64_t _startTimestamp;
     std::shared_ptr<SamplingProfiler> _profiler;
     SentryDebugImageProvider *_debugImageProvider;
+    thread::TIDType _mainThreadID;
 }
 
 - (instancetype)init
 {
+    if (![NSThread isMainThread]) {
+        [SentryLog logWithMessage:@"SentryProfiler must be initialized on the main thread"
+                         andLevel:kSentryLevelError];
+        return nil;
+    }
```

Comment:
`m`: Can we add a test for this? Trying to init from a background thread and checking if the returned value is `nil`?

Reply:
Will do!